File: CodeTest/Python/Programmers/타겟넘버.py
def solution(numbers, target):
    answer = 0
    sum_numbers = sum(numbers)
    
    for i in range(1<<len(numbers)):            # 각 요소가 음수 아니면 양수이므로 2가지 경우의 수
        tmp_sum = sum_numbers                   # 리스트의 합을 임시변수에 저장
        for j in range(len(numbers)):
            if i & (1<<j):
                tmp_sum += (numbers[j] * -2)    # 임시변수에 각 요소가 음수인 경우 합산
        if tmp_sum == target:                   # 타겟변수와 값이 같으면 카운트
            answer += 1
    
    return answer

File: CodeTest/JavaScript/programmers/타겟 넘버.js
function solution(numbers, target) {
  let answer = 0                      // 정답 변수
  
  function dfs(n, ans) {              // dfs 탐색 함수
    if (n >= numbers.length) {        // numbers 길이만큼 탐색 했다면
      if (ans === target) {           // 현재까지 정답이 타겟 숫자와 동일하면
          answer++                    // 정답 증가
      }
      return
    }    
      
    dfs(n + 1, ans + numbers[n])      // 현재 숫자를 더하기
    dfs(n + 1, ans - numbers[n])      // 현재 숫자를 빼기
  }
  
  dfs(0, 0)                           // dfs 탐색
      
  return answer;
}

console.log(solution([1, 1, 1, 1, 1], 3))
console.log(solution([4, 1, 2, 1], 4))